code/parachain/runtime/common/src/fees.rs

```diff
@@ -47,11 +48,16 @@ pub fn multi_existential_deposits<AssetsRegistry>(_currency_id: &CurrencyId) ->
 }
 
 #[cfg(not(feature = "runtime-benchmarks"))]
-pub fn multi_existential_deposits<AssetsRegistry: AssetRatioInspect<AssetId = CurrencyId>>(
+pub fn multi_existential_deposits<
+	AssetsRegistry: AssetRatioInspect<AssetId = CurrencyId>
+		+ AssetExistentialDepositInspect<AssetId = CurrencyId, Balance = Balance>,
+>(
 	currency_id: &CurrencyId,
 ) -> Balance {
-	PriceConverter::<AssetsRegistry>::to_asset_balance(NATIVE_EXISTENTIAL_DEPOSIT, *currency_id)
-		.unwrap_or(Balance::MAX)
+	AssetsRegistry::existential_deposit(*currency_id).unwrap_or_else(|_| {
+		PriceConverter::<AssetsRegistry>::to_asset_balance(NATIVE_EXISTENTIAL_DEPOSIT, *currency_id)
```

Comment:
Yes, `PriceConverter` could be implemented on assets registry. I wanted to keep this here for legacy assets, but assuming a successful migration, it would have no value here.

Reply:
e1b9e551a7646bddc53125f2195aaed890e0e5ca